main/core.js

```diff
@@ -50,13 +50,13 @@ async function setup (ctx) {
  * @param {Context} ctx
  */
 async function start (ctx) {
-  assert(wallet.getAddress(), 'Core requires FIL address')
+  assert(await wallet.getAddress(), 'Core requires FIL address')
   console.log('Starting Core...')
 
   const childProcess = fork(corePath, ['--json'], {
     env: {
       ...process.env,
-      FIL_WALLET_ADDRESS: wallet.getAddress(),
+      FIL_WALLET_ADDRESS: await wallet.getAddress(),
```

Comment:
Do we have to await `getAddress` call twice? Can that delay startup? Have you considered storing the address in a variable instead?

```js
async function start (ctx) {
  const walletAddress = await wallet.getAddress()
  assert(walletAddress, 'Core requires FIL address')
  // ...
     FIL_WALLET_ADDRESS: walletAddress
```

Reply:
good point!